src/IP.php

```diff
@@ -110,6 +110,10 @@ abstract class IP
      */
     public function __construct($ip)
     {
+        if ($ip === null) {
+            return;
```

Comment:
You are introducing a "backdoor" in the constructor, allowing for construction of invalid objects, and this isn't a good practice. Example:

```
$ip = new PhpIP\IPv4(null);
echo $ip;
```
now completely breaks with unhelpful errors:
> PHP Warning:  gmp_strval(): Unable to convert variable to GMP - wrong type
> PHP Fatal error:  Method PhpIP\IPv4::__toString() must not throw an exception, caught TypeError: Return value of PhpIP\IP::numeric() must be of the type string, bool returned

instead of a clean constructor exception (as currently):
> InvalidArgumentException: Unsupported argument type: "NULL".